src/server.ts

```diff
@@ -39,7 +39,20 @@ export class Server {
 
   async start() {
     this.setup()
-    new Promise<void>(res => this.server!.listen(this.port, res))
+    return new Promise<void>((resolve, reject) => {
```

Comment:
Fixed to `return` Promise instance. 

Current code causes "Unhandled promise rejection" because this Promise is not handled.